waiter/src/waiter/main.clj

```diff
@@ -56,6 +57,10 @@
             (log/error e "Unable to consume request stream"))))
       resp)))
 
+(defn- wrap-request-cookies [handler]
```

Comment:
The library already has `wrap-cookies`.

https://github.com/ring-clojure/ring/blob/master/ring-core/src/ring/middleware/cookies.clj#L137

Should we go all in and use that?  It also uses `cookies-response`.